src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -49,18 +55,20 @@ public BtConnectorHelper() {
         this.context = jxcore.activity.getBaseContext();
     }
 
-
-    public BTConnector.WifiBtStatus Start(String peerName,int port){
+    public synchronized BTConnector.WifiBtStatus Start(String peerName,int port){
         this.mServerPort = port;
         this.myPeerIdentifier= GetBluetoothAddress();
         this.myPeerName = peerName;
-        this.lastAvailableList.clear();
+        synchronized (lastAvailableList) {
```

Comment:
@DrJukka Just adding your tag sign so you get notified since this comment is in your code.